### ISSUE TITLE
Mismatch between range and instant queries

### ISSUE THREAD

**Issue by raags** [VictoriaMetrics/VictoriaMetrics#845]
**Describe the bug**
The following query results in a metric being missed in `query_range` v/s `query` API:
```
$ curl -s  'localhost:8428/api/v1/query?query=count+%28node_cpu_seconds_total%7Binstance%3D~%22%5Einstancefoo.%2A%22%7D%29+by+%28instance%29+%2A+0+%2B+101&time=1603129382'  | jq '.'  | grep metric | wc -l
$ 18
```
v/s query_range

```
$ curl -s  'localhost:8428/api/v1/query?query=count+%28node_cpu_seconds_total%7Binstance%3D~%22%5Einstancefoo.%2A%22%7D%29+by+%28instance%29+%2A+0+%2B+101&time=1603129382'  | jq '.'  | grep metric | wc -l
$ 19
```
I ended up checking the difference after noticing that vmalert was skipping an instance in the above query, as part of a recording rule.

**To Reproduce**
None as of now. Observed this behaviour with a single recording rule. I will have to confirm if this is happening with others as well.

**Expected behavior**
The `query` and `query_range` apis should not exclude any metric.

**Screenshots**
n/a

**Version**
The line returned when passing `--version` command line flag to binary. For example:
```
$ ./victoria-metrics-prod --version
victoria-metrics-20201013-140052-tags-v1.44.0-0-g94978af9b
```

**Used command-line flags**
```
/usr/local/bin/victoriametrics -loggerDisableTimestamps --storageDataPath=/var/lib/victoriametrics --selfScrapeInterval=0 --retentionPeriod=6 --memory.allowedPercent=60 --search.maxUniqueTimeseries=1000000
```

**Additional context**
No major changes in VM, other than the recent upgrade to v1.44.0.

**Comment by f41gh7** [comment]
Seems like it was broken at https://github.com/VictoriaMetrics/VictoriaMetrics/commit/9aa3b6576648ca23dba1c11ef213e91123203399

**Comment by valyala** [comment]
The issue must be fixed in the commit daacbc7e3462229d1afaeaec4e982ef4a9e6eb27

**Comment by valyala** [comment]
The bugfix has been included in [v1.45.0](https://github.com/VictoriaMetrics/VictoriaMetrics/releases). @raags , could you verify whether this release works as expected for your case?

**Comment by valyala** [comment]
An additional fix for this issue has been included in [v1.47.0](https://github.com/VictoriaMetrics/VictoriaMetrics/releases). @raags, could you verify that the issue is fixed there?

**Comment by raags** [comment]
I can confirm the issue is fixed.
